app/controllers/admin_controller.rb

```diff
@@ -0,0 +1,11 @@
+class AdminController < ApplicationController
```

Comment:
If I'm understanding this correctly you are creating a parent controller for all the admin controllers to inhering from.

I would consider using the namespace for this for a more idiomatic ruby approch
```
Admin::ApplicationController < ApplicationController
```

Reply:
Yep, so I can do things like handling authentication in a before action on all the admin controllers